
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>VAJA: Poravnava zaporedij NW &amp; SW (Python) &#8212; Biokemijska informatika</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!--
    this give us a css class that will be invisible only if js is disabled
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=8878045cc6db502f8baf" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=8878045cc6db502f8baf" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=87e54e7c" />
  
  <!-- So that users can add custom icons -->
  <script src="../_static/scripts/fontawesome.js?digest=8878045cc6db502f8baf"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=8878045cc6db502f8baf" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=8878045cc6db502f8baf" />

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=36754332"></script>
    <script async="async" src="https://www.googletagmanager.com/gtag/js?id=G-KG7KWKX5VG"></script>
    <script>
                window.dataLayer = window.dataLayer || [];
                function gtag(){ dataLayer.push(arguments); }
                gtag('js', new Date());
                gtag('config', 'G-KG7KWKX5VG');
            </script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>
                window.dataLayer = window.dataLayer || [];
                function gtag(){ dataLayer.push(arguments); }
                gtag('js', new Date());
                gtag('config', 'G-KG7KWKX5VG');
            </script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'vaje/poravnava2_nw_sw';</script>
    <link rel="canonical" href="https://mpavsic.github.io/biokeminfo/vaje/poravnava2_nw_sw.html" />
    <link rel="icon" href="../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="DEMO: Izračun lastne matrike in njena uporaba" href="poravnava2_lastna_matrika.html" />
    <link rel="prev" title="VAJA: Poravnava zaporedij NW &amp; SW (spletna orodja)" href="poravnava2_web.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="" />
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../naslovnica.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/logo.png" class="logo__image only-light" alt="Biokemijska informatika - Home"/>
    <img src="../_static/logo.png" class="logo__image only-dark pst-js-only" alt="Biokemijska informatika - Home"/>
  
  
</a></div>
        <div class="sidebar-primary-item">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../naslovnica.html">
                    Predgovor
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Zbirke</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../osnove/pubmed.html">PubMed</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="pubmed_web.html">VAJA: Iskanje po PubMed</a></li>
<li class="toctree-l2"><a class="reference internal" href="pubmed_eutilities.html">DEMO: Programski dostop do PubMed</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../osnove/nt.html">Nukleotidna zaporedja</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="nt-oligo_analiza.html">VAJA: Enostavna analiza nt-zaporedja (Python)</a></li>
<li class="toctree-l2"><a class="reference internal" href="genbank_web.html">VAJA: Dostop do GenBank v brskalniku</a></li>
<li class="toctree-l2"><a class="reference internal" href="genbank_eutilities.html">VAJA: Programski dostop do GenBank (Python)</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../osnove/ak.html">Aminokislinska zaporedja</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="uniprot_web.html">VAJA: Dostop do UniProt v brskalniku</a></li>
<li class="toctree-l2"><a class="reference internal" href="uniprot_rest.html">VAJA: Programski dostop do UniProt (Python)</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Poravnave</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../osnove/dotplot.html">Točkovni diagram</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="dotplot_web.html">VAJA: Točkovni diagram</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../osnove/poravnave.html">Poravnave</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="matrika_zamenjav.html">VAJA: Izračun matrike zamenjav (Python)</a></li>
<li class="toctree-l2"><a class="reference internal" href="poravnava2_web.html">VAJA: Poravnava zaporedij NW &amp; SW (spletna orodja)</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">VAJA: Poravnava zaporedij NW &amp; SW (Python)</a></li>
<li class="toctree-l2"><a class="reference internal" href="poravnava2_lastna_matrika.html">DEMO: Izračun lastne matrike in njena uporaba</a></li>
<li class="toctree-l2"><a class="reference internal" href="blast.html">VAJA: Iskanje podobnih zaporedij z BLAST</a></li>
<li class="toctree-l2"><a class="reference internal" href="msa.html">VAJA: Poravnave več zaporedij</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Grafi</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../osnove/grafi.html">Grafi v molekularni biologiji</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="grafi_cytoscape.html">VAJA: Grafi in program Cytoscape</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Strukture</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../osnove/strukture.html">Strukture</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="podobne_strukture.html">VAJA: Iskanje podobnih struktur</a></li>
<li class="toctree-l2"><a class="reference internal" href="modeliranje.html">VAJA: Modeliranje na osnovi homologije</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Priloge</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../priloge/programi.html">Programi</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../priloge/python.html">Python</a></li>
<li class="toctree-l2"><a class="reference internal" href="../priloge/jupyterlab.html">JupyterLab</a></li>
<li class="toctree-l2"><a class="reference internal" href="../priloge/markdown.html">Markdown</a></li>
<li class="toctree-l2"><a class="reference internal" href="../priloge/git.html">Git in GitHub</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../priloge/datoteke.html">Datoteke</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../priloge/datoteke_formati.html">Pogosti formati datotek</a></li>
<li class="toctree-l2"><a class="reference internal" href="../priloge/datoteke_poimenovanje.html">Poimenovanje datotek</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Izvedba vaj</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../izvedba/vaje_2024-2025.html">Vaje 2024/2025</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Seminar</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../seminar/seminar.html">Splošno o seminarju</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../seminar/seminar_2023-2024/seminar.html">Seminar 2023/2024</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../seminar/seminar_2023-2024/seminar-predloga.html">Predloga</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../seminar/seminar_2021-2022/seminar.html">Seminar 2021/2022</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../seminar/seminar_2021-2022/navodila.html">Navodila</a></li>
<li class="toctree-l2"><a class="reference internal" href="../seminar/seminar_2021-2022/seminar-predloga.html">Predloga</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Razno</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../razno/zanimivosti.html">Zanimivosti</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
      <div class="sidebar-primary-item">
<div id="ethical-ad-placement"
      class="flat"
      data-ea-publisher="readthedocs"
      data-ea-type="readthedocs-sidebar"
      data-ea-manual="true">
</div></div>
  </div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-launch-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Launch interactive content">
    <i class="fas fa-rocket"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://mybinder.org/v2/gh/mpavsic/biokeminfo/main?urlpath=lab/tree/biokeminfo/vaje/poravnava2_nw_sw.ipynb" target="_blank"
   class="btn btn-sm dropdown-item"
   title="Launch on Binder"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  
    <img alt="Binder logo" src="../_static/images/logo_binder.svg">
  </span>
<span class="btn__text-container">Binder</span>
</a>
</li>
      
      
      
      
      <li><a href="https://colab.research.google.com/github/mpavsic/biokeminfo/blob/main/biokeminfo/vaje/poravnava2_nw_sw.ipynb" target="_blank"
   class="btn btn-sm dropdown-item"
   title="Launch on Colab"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  
    <img alt="Colab logo" src="../_static/images/logo_colab.png">
  </span>
<span class="btn__text-container">Colab</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/mpavsic/biokeminfo" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/mpavsic/biokeminfo/edit/main/biokeminfo/vaje/poravnava2_nw_sw.ipynb" target="_blank"
   class="btn btn-sm btn-source-edit-button dropdown-item"
   title="Suggest edit"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-pencil-alt"></i>
  </span>
<span class="btn__text-container">Suggest edit</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/mpavsic/biokeminfo/issues/new?title=Issue%20on%20page%20%2Fvaje/poravnava2_nw_sw.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/vaje/poravnava2_nw_sw.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button>


<button class="btn btn-sm pst-navbar-icon search-button search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
</button>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>VAJA: Poravnava zaporedij NW & SW (Python)</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#zgledi">Zgledi</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#definicija-zaporedja">Definicija zaporedja</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#izbor-matrike-zamenjav">Izbor matrike zamenjav</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#globalna-poravnava">Globalna poravnava</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#lokalna-poravnava">Lokalna poravnava</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#globalna-in-lokalna-poravnava-aminokislinskih-zaporedij">Globalna in lokalna poravnava aminokislinskih zaporedij</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#naloga">Naloga</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="vaja-poravnava-zaporedij-nw-sw-python">
<h1>VAJA: Poravnava zaporedij NW &amp; SW (Python)<a class="headerlink" href="#vaja-poravnava-zaporedij-nw-sw-python" title="Link to this heading">#</a></h1>
<p>V Pythonu lahko z uporabo paketa <a class="reference external" href="https://biopython.org">BioPython</a> na enostaven način delamo poravnave aminokislinskih zaporedij. Uporaba algoritmov, ki dajo matematično optimalno poravnavo (npr. <strong>Needleman-Wunsch</strong> in <strong>Smith-Waterman</strong>), je v primeru manjšega obsega vhodnih podatkov popolnoma normalno izvedljiva na domačem računalniku. V primeru daljšega računanja se lahko uporabi (“pokliče” iz Pythona) orodja na npr. strežniku EMBOSS (opis API-ja za to je tukaj je v <a class="reference external" href="http://europepmc.org/article/MED/30976793">članku</a>).</p>
<p>Mi si bomo pogledali enostavne primere poravnave, kjer lahko poljubno variiramo parametre (npr. uporabljena matrika zamenjav ter kazni za vrzeli) in opazujemo, kakšne poravnave dobimo. Kot pri preteklih vajah so spodaj že pripravljeni zgledi kode, ki s spremljajočim besedilom ilustrirajo uporabo, na dnu datoteke pa je enostavna naloga, ki ilustrira možnost variacije parametrov.</p>
<hr class="docutils" />
<section id="zgledi">
<h2>Zgledi<a class="headerlink" href="#zgledi" title="Link to this heading">#</a></h2>
<section id="definicija-zaporedja">
<h3>Definicija zaporedja<a class="headerlink" href="#definicija-zaporedja" title="Link to this heading">#</a></h3>
<p>Najprej bomo <strong>definirali objekt, ki vsebuje zaporedje</strong>, oz. dva taka objekta. To lahko naredimo na več načinov, eden je direktna ustvaritev, kjer sami podamo zaporedje. Zaporedja lahko preberemo tudi iz <em>lokalno shranjene datoteke</em> v različnih formatih (npr. Genbank ali FASTA) ter seveda tudi iz <em>spletnih podatkovnih zbirk (npr. Genbank ali UniProt)</em>, kar smo si že pogledali na prejšnjih vajah.</p>
<p>Torej, objekt definiramo podobno kot pri eni izmed začetnih vaj z BioPython:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">Bio.Seq</span><span class="w"> </span><span class="kn">import</span> <span class="n">Seq</span>
<span class="n">seq1</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s1">&#39;AAKCLVMKAEMNGSKLGRRAKPEGALQNNDGLYDPDCDESGLFKAKQCNGTSMCWCVNTAGVRRTDKDTEITC&#39;</span><span class="p">)</span>
<span class="n">seq2</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s1">&#39;KTRCQLEREHILGAAGGADAQRPTLQGMFVPQCDEYGHYVPTQCHHSTGYCWCVDRDGRELEGSRTPPGMRPPC&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Lahko smo še malce bolj kreativni in zraven zaporedja samega definiramo še kakšne pripise (<a class="reference external" href="https://biopython.org/wiki/SeqRecord">dokumentacija</a>):</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">Bio.Seq</span><span class="w"> </span><span class="kn">import</span> <span class="n">Seq</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">Bio.SeqRecord</span><span class="w"> </span><span class="kn">import</span> <span class="n">SeqRecord</span>
<span class="n">seq1</span> <span class="o">=</span> <span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s1">&#39;AAKCLVMKAEMNGSKLGRRAKPEGALQNNDGLYDPDCDESGLFKAKQCNGTSMCWCVNTAGVRRTDKDTEITC&#39;</span><span class="p">),</span>
                 <span class="nb">id</span><span class="o">=</span><span class="s1">&#39;P16422&#39;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;EpCAM&#39;</span><span class="p">,</span>
                 <span class="n">description</span><span class="o">=</span><span class="s1">&#39;epithelial cell adhesion molecule, 63-135&#39;</span><span class="p">)</span>
<span class="n">seq2</span> <span class="o">=</span> <span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s1">&#39;LTKCQEEVSHIPAVHPGSFRPKCDENGNYLPLQCYGSIGYCWCVFPNGTEVPNTRSRGHHNC&#39;</span><span class="p">),</span>
                 <span class="nb">id</span><span class="o">=</span><span class="s1">&#39;P04233&#39;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;HG2A&#39;</span><span class="p">,</span>
                 <span class="n">description</span><span class="o">=</span><span class="s1">&#39;HLA class II histocompatibility antigen gamma chain, 210-271&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">seq1</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">seq1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>ID: P16422
Name: EpCAM
Description: epithelial cell adhesion molecule, 63-135
Number of features: 0
Seq(&#39;AAKCLVMKAEMNGSKLGRRAKPEGALQNNDGLYDPDCDESGLFKAKQCNGTSMC...ITC&#39;)
ID: P16422
Name: EpCAM
Description: epithelial cell adhesion molecule, 63-135
Number of features: 0
Seq(&#39;AAKCLVMKAEMNGSKLGRRAKPEGALQNNDGLYDPDCDESGLFKAKQCNGTSMC...ITC&#39;)
</pre></div>
</div>
</div>
</div>
<p>Analogno bi lahko naredili tudi za DNA.</p>
<p>V primeru, da zaporedje oz. objekt definiramo na zgornji način, izpišemo zaporedje sámo tako:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="n">seq1</span><span class="o">.</span><span class="n">seq</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AAKCLVMKAEMNGSKLGRRAKPEGALQNNDGLYDPDCDESGLFKAKQCNGTSMCWCVNTAGVRRTDKDTEITC
</pre></div>
</div>
</div>
</div>
</section>
<hr class="docutils" />
<section id="izbor-matrike-zamenjav">
<h3>Izbor matrike zamenjav<a class="headerlink" href="#izbor-matrike-zamenjav" title="Link to this heading">#</a></h3>
<p>Naslednje, kar zraven zaporedja potrebujemo za izračun poravnave, je <strong>substitucijska matrika</strong> (matrika zamenjav, <em>substitution matrix</em>). Dober nabor le-teh smo že dobili preko BioPythona, slednji pa nam tudi omogoča definicijo čisto poljubne matrike (matrike so definirane kot slovarji - <em>dictionary</em>).</p>
<p>Pa poglejmo, kaj imamo na voljo! Seznam matrik je dostopen preko tega spletnega naslova: <a class="reference external" href="https://biopython.org/DIST/docs/api/Bio.SubsMat.MatrixInfo-module.html">https://biopython.org/DIST/docs/api/Bio.SubsMat.MatrixInfo-module.html</a></p>
<p>Lahko pa tudi samo pokukamo, kaj imamo v trenutno nainstalirani verziji BioPythona na razpolago:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">Bio.Align</span><span class="w"> </span><span class="kn">import</span> <span class="n">substitution_matrices</span>
<span class="n">matrix_list</span> <span class="o">=</span> <span class="n">substitution_matrices</span><span class="o">.</span><span class="n">load</span><span class="p">()</span>
<span class="nb">print</span><span class="p">(</span><span class="n">matrix_list</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[&#39;BENNER22&#39;, &#39;BENNER6&#39;, &#39;BENNER74&#39;, &#39;BLASTN&#39;, &#39;BLASTP&#39;, &#39;BLOSUM45&#39;, &#39;BLOSUM50&#39;, &#39;BLOSUM62&#39;, &#39;BLOSUM80&#39;, &#39;BLOSUM90&#39;, &#39;DAYHOFF&#39;, &#39;FENG&#39;, &#39;GENETIC&#39;, &#39;GONNET1992&#39;, &#39;HOXD70&#39;, &#39;JOHNSON&#39;, &#39;JONES&#39;, &#39;LEVIN&#39;, &#39;MCLACHLAN&#39;, &#39;MDM78&#39;, &#39;MEGABLAST&#39;, &#39;NUC.4.4&#39;, &#39;PAM250&#39;, &#39;PAM30&#39;, &#39;PAM70&#39;, &#39;RAO&#39;, &#39;RISLER&#39;, &#39;SCHNEIDER&#39;, &#39;STR&#39;, &#39;TRANS&#39;]
</pre></div>
</div>
</div>
</div>
<p>Izberemo matriko ter jo izpišemo, da vidimo, kako je pravzaprav zapisana (spodaj ne razširjen zapis kode):</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">Bio.Align</span><span class="w"> </span><span class="kn">import</span> <span class="n">substitution_matrices</span>
<span class="n">matrix_name</span> <span class="o">=</span> <span class="s1">&#39;BLOSUM62&#39;</span>
<span class="n">matrix</span> <span class="o">=</span> <span class="n">substitution_matrices</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">matrix_name</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">matrix</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>#  Matrix made by matblas from blosum62.iij
#  * column uses minimum score
#  BLOSUM Clustered Scoring Matrix in 1/2 Bit Units
#  Blocks Database = /data/blocks_5.0/blocks.dat
#  Cluster Percentage: &gt;= 62
#  Entropy =   0.6979, Expected =  -0.5209
     A    R    N    D    C    Q    E    G    H    I    L    K    M    F    P    S    T    W    Y    V    B    Z    X    *
A  4.0 -1.0 -2.0 -2.0  0.0 -1.0 -1.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  1.0  0.0 -3.0 -2.0  0.0 -2.0 -1.0  0.0 -4.0
R -1.0  5.0  0.0 -2.0 -3.0  1.0  0.0 -2.0  0.0 -3.0 -2.0  2.0 -1.0 -3.0 -2.0 -1.0 -1.0 -3.0 -2.0 -3.0 -1.0  0.0 -1.0 -4.0
N -2.0  0.0  6.0  1.0 -3.0  0.0  0.0  0.0  1.0 -3.0 -3.0  0.0 -2.0 -3.0 -2.0  1.0  0.0 -4.0 -2.0 -3.0  3.0  0.0 -1.0 -4.0
D -2.0 -2.0  1.0  6.0 -3.0  0.0  2.0 -1.0 -1.0 -3.0 -4.0 -1.0 -3.0 -3.0 -1.0  0.0 -1.0 -4.0 -3.0 -3.0  4.0  1.0 -1.0 -4.0
C  0.0 -3.0 -3.0 -3.0  9.0 -3.0 -4.0 -3.0 -3.0 -1.0 -1.0 -3.0 -1.0 -2.0 -3.0 -1.0 -1.0 -2.0 -2.0 -1.0 -3.0 -3.0 -2.0 -4.0
Q -1.0  1.0  0.0  0.0 -3.0  5.0  2.0 -2.0  0.0 -3.0 -2.0  1.0  0.0 -3.0 -1.0  0.0 -1.0 -2.0 -1.0 -2.0  0.0  3.0 -1.0 -4.0
E -1.0  0.0  0.0  2.0 -4.0  2.0  5.0 -2.0  0.0 -3.0 -3.0  1.0 -2.0 -3.0 -1.0  0.0 -1.0 -3.0 -2.0 -2.0  1.0  4.0 -1.0 -4.0
G  0.0 -2.0  0.0 -1.0 -3.0 -2.0 -2.0  6.0 -2.0 -4.0 -4.0 -2.0 -3.0 -3.0 -2.0  0.0 -2.0 -2.0 -3.0 -3.0 -1.0 -2.0 -1.0 -4.0
H -2.0  0.0  1.0 -1.0 -3.0  0.0  0.0 -2.0  8.0 -3.0 -3.0 -1.0 -2.0 -1.0 -2.0 -1.0 -2.0 -2.0  2.0 -3.0  0.0  0.0 -1.0 -4.0
I -1.0 -3.0 -3.0 -3.0 -1.0 -3.0 -3.0 -4.0 -3.0  4.0  2.0 -3.0  1.0  0.0 -3.0 -2.0 -1.0 -3.0 -1.0  3.0 -3.0 -3.0 -1.0 -4.0
L -1.0 -2.0 -3.0 -4.0 -1.0 -2.0 -3.0 -4.0 -3.0  2.0  4.0 -2.0  2.0  0.0 -3.0 -2.0 -1.0 -2.0 -1.0  1.0 -4.0 -3.0 -1.0 -4.0
K -1.0  2.0  0.0 -1.0 -3.0  1.0  1.0 -2.0 -1.0 -3.0 -2.0  5.0 -1.0 -3.0 -1.0  0.0 -1.0 -3.0 -2.0 -2.0  0.0  1.0 -1.0 -4.0
M -1.0 -1.0 -2.0 -3.0 -1.0  0.0 -2.0 -3.0 -2.0  1.0  2.0 -1.0  5.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0  1.0 -3.0 -1.0 -1.0 -4.0
F -2.0 -3.0 -3.0 -3.0 -2.0 -3.0 -3.0 -3.0 -1.0  0.0  0.0 -3.0  0.0  6.0 -4.0 -2.0 -2.0  1.0  3.0 -1.0 -3.0 -3.0 -1.0 -4.0
P -1.0 -2.0 -2.0 -1.0 -3.0 -1.0 -1.0 -2.0 -2.0 -3.0 -3.0 -1.0 -2.0 -4.0  7.0 -1.0 -1.0 -4.0 -3.0 -2.0 -2.0 -1.0 -2.0 -4.0
S  1.0 -1.0  1.0  0.0 -1.0  0.0  0.0  0.0 -1.0 -2.0 -2.0  0.0 -1.0 -2.0 -1.0  4.0  1.0 -3.0 -2.0 -2.0  0.0  0.0  0.0 -4.0
T  0.0 -1.0  0.0 -1.0 -1.0 -1.0 -1.0 -2.0 -2.0 -1.0 -1.0 -1.0 -1.0 -2.0 -1.0  1.0  5.0 -2.0 -2.0  0.0 -1.0 -1.0  0.0 -4.0
W -3.0 -3.0 -4.0 -4.0 -2.0 -2.0 -3.0 -2.0 -2.0 -3.0 -2.0 -3.0 -1.0  1.0 -4.0 -3.0 -2.0 11.0  2.0 -3.0 -4.0 -3.0 -2.0 -4.0
Y -2.0 -2.0 -2.0 -3.0 -2.0 -1.0 -2.0 -3.0  2.0 -1.0 -1.0 -2.0 -1.0  3.0 -3.0 -2.0 -2.0  2.0  7.0 -1.0 -3.0 -2.0 -1.0 -4.0
V  0.0 -3.0 -3.0 -3.0 -1.0 -2.0 -2.0 -3.0 -3.0  3.0  1.0 -2.0  1.0 -1.0 -2.0 -2.0  0.0 -3.0 -1.0  4.0 -3.0 -2.0 -1.0 -4.0
B -2.0 -1.0  3.0  4.0 -3.0  0.0  1.0 -1.0  0.0 -3.0 -4.0  0.0 -3.0 -3.0 -2.0  0.0 -1.0 -4.0 -3.0 -3.0  4.0  1.0 -1.0 -4.0
Z -1.0  0.0  0.0  1.0 -3.0  3.0  4.0 -2.0  0.0 -3.0 -3.0  1.0 -1.0 -3.0 -1.0  0.0 -1.0 -3.0 -2.0 -2.0  1.0  4.0 -1.0 -4.0
X  0.0 -1.0 -1.0 -1.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -2.0  0.0  0.0 -2.0 -1.0 -1.0 -1.0 -1.0 -1.0 -4.0
* -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0 -4.0  1.0
</pre></div>
</div>
</div>
</div>
<p>Drug način je takle, dobimo enako (izpis je izključen):</p>
<p>Če nas zanima neka konretna vrednost iz matrike, na primer vrednost za substitucijo valina in izolevcina, lahko to naredimo na prav enostaven način:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="n">matrix</span><span class="p">[</span><span class="s1">&#39;V&#39;</span><span class="p">,</span> <span class="s1">&#39;I&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3.0
</pre></div>
</div>
</div>
</div>
<p>Dobro, to je bilo nekaj uvoda, sedaj pa gremo k stvari - poravnavam!</p>
</section>
<hr class="docutils" />
<section id="globalna-poravnava">
<h3>Globalna poravnava<a class="headerlink" href="#globalna-poravnava" title="Link to this heading">#</a></h3>
<p>Za poravnavo zaporedij imamo v <a class="reference external" href="https://biopython.org">BioPython</a> na voljo več orodij. Eno izmed njih, modul <strong><a class="reference external" href="https://biopython.org/DIST/docs/api/Bio.pairwise2-module.html">pairwise2</a></strong>, omogoča izračun poravnave dveh zaporedij z dinamičnim programiranjem, med drugim preko algoritmov Needleman-Wunsch in Smith-Waterman.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># pairwise2 lahko definiramo kot aln, da vsakič ne izpisujemo pairwise2</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">Bio</span><span class="w"> </span><span class="kn">import</span> <span class="n">pairwise2</span> <span class="k">as</span> <span class="n">aln</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/opt/homebrew/Caskroom/miniforge/base/envs/teach/lib/python3.12/site-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.
  warnings.warn(
</pre></div>
</div>
</div>
</div>
<p>Za ilustracijo bomo poravnavo izvedli na nt-zaporedjih. Pa jih definirajmo, enostavno, kot <em>string</em>:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">seq1</span> <span class="o">=</span> <span class="s1">&#39;AGCCGA&#39;</span>
<span class="n">seq2</span> <span class="o">=</span> <span class="s1">&#39;ACTA&#39;</span>
</pre></div>
</div>
</div>
</div>
<p>Globalno poravnavo (ali več njih) dobimo tako:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">alignments</span> <span class="o">=</span> <span class="n">aln</span><span class="o">.</span><span class="n">align</span><span class="o">.</span><span class="n">globalxx</span><span class="p">(</span><span class="n">seq1</span><span class="p">,</span> <span class="n">seq2</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">alignments</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[Alignment(seqA=&#39;AGCCG-A&#39;, seqB=&#39;A--C-TA&#39;, score=3.0, start=0, end=7), Alignment(seqA=&#39;AGCCG-A&#39;, seqB=&#39;A-C--TA&#39;, score=3.0, start=0, end=7), Alignment(seqA=&#39;AGCCGA&#39;, seqB=&#39;A--CTA&#39;, score=3.0, start=0, end=6), Alignment(seqA=&#39;AGCCGA&#39;, seqB=&#39;A-C-TA&#39;, score=3.0, start=0, end=6), Alignment(seqA=&#39;AGCCGA&#39;, seqB=&#39;A-CT-A&#39;, score=3.0, start=0, end=6)]
</pre></div>
</div>
</div>
</div>
<p>Na ta način dobimo v <code class="docutils literal notranslate"><span class="pre">poravnave</span></code> seznam (<em>list</em>) poravnav, od katerih je vsaka predstavljena kot terka (<em>tuple</em>), sestavljena iz prvega in drugega zaporedja, sledijo pa tri številke:</p>
<ul class="simple">
<li><p>vrednost poravnave (<em>score</em> kot float),</p></li>
<li><p>začetno mesto poravnave (štejemo od 0),</p></li>
<li><p>končno mesto poravnave.</p></li>
</ul>
<p>Ime funkcije, v našem primeru <code class="docutils literal notranslate"><span class="pre">globalxx</span></code>, je dejansko sestavljeno iz 3 komponent oblike <code class="docutils literal notranslate"><span class="pre">&lt;alignment</span> <span class="pre">type&gt;XX</span></code>. Prvi parameter oz. ime samo določa vrsto poravnave (<code class="docutils literal notranslate"><span class="pre">global</span></code> ali  <code class="docutils literal notranslate"><span class="pre">local</span></code> za globalno oz. lokalno poravnavo), druga parametra pa določata, kako se upoštevajo ujemanja in neujemanja (prvi znak) ter vrzeli (drugi znak). Pomen teh stikal je podrobneje opisan v <a class="reference external" href="https://biopython.org/docs/1.77/api/Bio.pairwise2.html">dokumentaciji</a>.</p>
<p>Možnosti za ujemanja/neujemanja:</p>
<div class="pst-scrollable-table-container"><table class="table">
<thead>
<tr class="row-odd"><th class="head text-left"><p>vrednost</p></th>
<th class="head text-left"><p>pomen</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-left"><p>x</p></td>
<td class="text-left"><p>brez posebnih vrednosti (ujemanje = 1, neujemanje = 0)</p></td>
</tr>
<tr class="row-odd"><td class="text-left"><p>m</p></td>
<td class="text-left"><p>ujemanje je vrednost za identične znake, sicer gre za neujemanje</p></td>
</tr>
<tr class="row-even"><td class="text-left"><p>d</p></td>
<td class="text-left"><p>vrednost za posamezni par znakov se prebere iz slovarja</p></td>
</tr>
<tr class="row-odd"><td class="text-left"><p>c</p></td>
<td class="text-left"><p>vrednosti da callback funkcija</p></td>
</tr>
</tbody>
</table>
</div>
<p>Množnosti za vrzeli:</p>
<div class="pst-scrollable-table-container"><table class="table">
<thead>
<tr class="row-odd"><th class="head text-left"><p>vrednost</p></th>
<th class="head text-left"><p>pomen</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-left"><p>x</p></td>
<td class="text-left"><p>brez kazni za vrzeli (oz. kazen je enaka 0)</p></td>
</tr>
<tr class="row-odd"><td class="text-left"><p>s</p></td>
<td class="text-left"><p>enake kazni za odprtje in razširitev vrzeli za obe zaporedji</p></td>
</tr>
<tr class="row-even"><td class="text-left"><p>d</p></td>
<td class="text-left"><p>različne kazni za odprtje in razširitev vrzeli za posamezni zaporedji</p></td>
</tr>
<tr class="row-odd"><td class="text-left"><p>c</p></td>
<td class="text-left"><p>vrednosti da callback funkcija</p></td>
</tr>
</tbody>
</table>
</div>
<p>Vrednost poravnave je vedno izračunana po naslednji enačbi:
$<span class="math notranslate nohighlight">\(S = \sum s_{a,b} + \sum (I + (n-1) E)\)</span>$</p>
<p>pri čemer je <span class="math notranslate nohighlight">\(s_{a,b}\)</span> vrednost iz matrike zamenjav (substitucijske matrike), <span class="math notranslate nohighlight">\(I\)</span> in <span class="math notranslate nohighlight">\(E\)</span> pa kazni za odprtje oz. razširitev vrzeli.</p>
<p>V bolj pregledno obliko lahko poravnavo preoblikujemo z uporabo <code class="docutils literal notranslate"><span class="pre">format_alignment</span></code> (pozor - spodaj je prikazano, kako izpišemo vse poravnave, ki smo jih dobili - vse imajo enako vrednosti):</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">Bio.pairwise2</span><span class="w"> </span><span class="kn">import</span> <span class="n">format_alignment</span>
<span class="k">for</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">alignments</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">format_alignment</span><span class="p">(</span><span class="o">*</span><span class="n">a</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AGCCG-A
|  |  |
A--C-TA
  Score=3

AGCCG-A
| |   |
A-C--TA
  Score=3

AGCCGA
|  |.|
A--CTA
  Score=3

AGCCGA
| | .|
A-C-TA
  Score=3

AGCCGA
| |. |
A-CT-A
  Score=3
</pre></div>
</div>
</div>
</div>
<p>Če bi želeli vrednosti za ujemanje in neujemanje posebej nastaviti (opis je nekaj polj višje), lahko to naredimo na način, prikazan spodaj. Bodite pozorni, kako je koda zapisana tokrat - v obliki zanke <code class="docutils literal notranslate"><span class="pre">for</span></code>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">alignments</span> <span class="o">=</span> <span class="n">aln</span><span class="o">.</span><span class="n">align</span><span class="o">.</span><span class="n">globalmx</span><span class="p">(</span><span class="n">seq1</span><span class="p">,</span> <span class="n">seq2</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">)</span>
<span class="k">for</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">alignments</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">format_alignment</span><span class="p">(</span><span class="o">*</span><span class="n">a</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AGCCG-A
|  |  |
A--C-TA
  Score=6

AGCCG-A
| |   |
A-C--TA
  Score=6
</pre></div>
</div>
</div>
</div>
<p>Še en primer - določimo kazen za odprtje in razširitev vrzeli. Uporabimo lahko tudi ključne besede, kar pomaga pri razumevanju, kaj je kaj.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">alignments</span> <span class="o">=</span> <span class="n">aln</span><span class="o">.</span><span class="n">align</span><span class="o">.</span><span class="n">globalms</span><span class="p">(</span><span class="n">seq1</span><span class="p">,</span> <span class="n">seq2</span><span class="p">,</span> <span class="n">match</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">mismatch</span><span class="o">=-</span><span class="mi">1</span><span class="p">,</span> <span class="nb">open</span><span class="o">=-</span><span class="mi">10</span><span class="p">,</span> <span class="n">extend</span><span class="o">=-</span><span class="mi">5</span><span class="p">)</span>
<span class="k">for</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">alignments</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">format_alignment</span><span class="p">(</span><span class="o">*</span><span class="n">a</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AGCCGA
|  |.|
A--CTA
  Score=-10
</pre></div>
</div>
</div>
</div>
<p>Izpis lahko tudi prilagodimo - več o tem lahko preberemo v <a class="reference external" href="https://biopython.org/DIST/docs/api/Bio.pairwise2-module.html">opisu modula pairwise2</a>. Spodaj je primer.</p>
<p>Lahko izpišemo zgolj npr. vrednost poravnave (prikazano je za prvo poravnavo v seznamu poravnav):</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="n">alignments</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">score</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>-10.0
</pre></div>
</div>
</div>
</div>
<p>Pri samem izračunu poravnave lahko nastavimo še nekatere parametre kot je (vir: <a class="reference external" href="https://biopython.org/DIST/docs/api/Bio.pairwise2-module.html">opis modula pairwise2</a>):</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">penalize_extend_when_opening</span></code>: boolean (default: False) - ali naj bo odprtje vrzeli dodatno kaznovano še kot ena razširitev ali ne (false = vrzel dolžine 1 je kaznovana samo s kaznijo za odprtje)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">penalize_end_gap</span></code>: boolean - ali naj kaznujemo končne vrzeli, privzeto je vključeno pri globalni poravnavi, pri lokalni to odpade</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">gap_char</span></code>: string (default: ‘-’) - kakšen znak naj se uporabi za vrzel</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">force_generic</span></code>: boolean (default: False) - če vklopimo, se bo uporabil generičen neoptimiziran algoritem za dinamično programiranje (počasno!)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">score_only</span></code>: boolean (default: False) - vrni samo vrednost poravnave brez poravnave same</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">one_alignment_only</span></code>: boolean (default: False) - vrni samo eno poravnavo</p></li>
</ul>
<p>Primer:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">alignments</span> <span class="o">=</span> <span class="n">aln</span><span class="o">.</span><span class="n">align</span><span class="o">.</span><span class="n">globalxx</span><span class="p">(</span><span class="n">seq1</span><span class="p">,</span> <span class="n">seq2</span><span class="p">,</span> <span class="n">gap_char</span><span class="o">=</span><span class="s1">&#39;Š&#39;</span><span class="p">,</span> <span class="n">one_alignment_only</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">Bio.pairwise2</span><span class="w"> </span><span class="kn">import</span> <span class="n">format_alignment</span>
<span class="nb">print</span><span class="p">(</span><span class="n">format_alignment</span><span class="p">(</span><span class="o">*</span><span class="n">alignments</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AGCCGŠA
|..|..|
AŠŠCŠTA
  Score=3
</pre></div>
</div>
</div>
</div>
<p>Tu je recimo ilustrirano, kako dobimo samo vrednost poravnave:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">alignments</span> <span class="o">=</span> <span class="n">aln</span><span class="o">.</span><span class="n">align</span><span class="o">.</span><span class="n">globalxx</span><span class="p">(</span><span class="n">seq1</span><span class="p">,</span> <span class="n">seq2</span><span class="p">,</span> <span class="n">score_only</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">alignments</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3.0
</pre></div>
</div>
</div>
</div>
<p>Že iz prejšnjega prikaza, še bolj nazorno pa iz preoblikovanega, lahko vidimo, da imamo dve poravnavi - ena ima blizu 3’-konca vrzel v obeh zaporedjih, druga pa na tem mestu neujemanje. Vrednost obeh je enaka, saj je kazen tako za odprtje vrzeli kot vrednost za neujemanje enaka 0.</p>
<p>BioPython sicer ne vsebuje substitucijskih matrik za nukleotidna zaporedja, tako da bi relevantno matriko morali sami uvoziti in nato uporabiti kot <em>dictionary</em>. Tega na tem mestu ne bomo počeli.</p>
</section>
<hr class="docutils" />
<section id="lokalna-poravnava">
<h3>Lokalna poravnava<a class="headerlink" href="#lokalna-poravnava" title="Link to this heading">#</a></h3>
<p>Poglejmo si še primer lokalne poravnave:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">seq3</span> <span class="o">=</span> <span class="s1">&#39;AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG&#39;</span>
<span class="n">seq4</span> <span class="o">=</span> <span class="s1">&#39;ACTCCGCTT&#39;</span>
<span class="n">alignments_global</span> <span class="o">=</span> <span class="n">aln</span><span class="o">.</span><span class="n">align</span><span class="o">.</span><span class="n">globalxx</span><span class="p">(</span><span class="n">seq3</span><span class="p">,</span> <span class="n">seq4</span><span class="p">)</span>
<span class="k">for</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">alignments_global</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">format_alignment</span><span class="p">(</span><span class="o">*</span><span class="n">a</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG
          ||| | |        |||    |     
----------ACT-C-C--------GCT----T-----
  Score=9

AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG
        |  || | |        |||    |     
--------A--CT-C-C--------GCT----T-----
  Score=9

AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG
       |   || | |        |||    |     
-------A---CT-C-C--------GCT----T-----
  Score=9

AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG
     |     || | |        |||    |     
-----A-----CT-C-C--------GCT----T-----
  Score=9

AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG
|          || | |        |||    |     
A----------CT-C-C--------GCT----T-----
  Score=9

AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG
          ||||  |        |||    |     
----------ACTC--C--------GCT----T-----
  Score=9

AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG
        |  |||  |        |||    |     
--------A--CTC--C--------GCT----T-----
  Score=9

AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG
       |   |||  |        |||    |     
-------A---CTC--C--------GCT----T-----
  Score=9

AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG
     |     |||  |        |||    |     
-----A-----CTC--C--------GCT----T-----
  Score=9

AGGGGATAAGACTCCGCTTAGAATAGCTAGAATAGCCG
|          |||  |        |||    |     
A----------CTC--C--------GCT----T-----
  Score=9
</pre></div>
</div>
</div>
</div>
<p>Predvidevam, da z zgornjimi rezultati nismo zadovoljni. Poizkusimo drugače, določimo kazni za vrzeli:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">alignments_local</span> <span class="o">=</span> <span class="n">aln</span><span class="o">.</span><span class="n">align</span><span class="o">.</span><span class="n">localxs</span><span class="p">(</span><span class="n">seq3</span><span class="p">,</span> <span class="n">seq4</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">,</span> <span class="mi">0</span><span class="p">)</span>
<span class="k">for</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">alignments_local</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">format_alignment</span><span class="p">(</span><span class="o">*</span><span class="n">a</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>11 ACTCCGCTT
   |||||||||
 1 ACTCCGCTT
  Score=9
</pre></div>
</div>
</div>
</div>
<p>Naj bo nukleotidnih zaporedij zaenkrat dovolj, gremo na aminokislinska zaporedja!</p>
</section>
<section id="globalna-in-lokalna-poravnava-aminokislinskih-zaporedij">
<h3>Globalna in lokalna poravnava aminokislinskih zaporedij<a class="headerlink" href="#globalna-in-lokalna-poravnava-aminokislinskih-zaporedij" title="Link to this heading">#</a></h3>
<p>Najprej definirajmo naši zaporedji (to je kopija kode od zgoraj):</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">Bio.Seq</span><span class="w"> </span><span class="kn">import</span> <span class="n">Seq</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">Bio.SeqRecord</span><span class="w"> </span><span class="kn">import</span> <span class="n">SeqRecord</span>
<span class="n">seq1</span> <span class="o">=</span> <span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s1">&#39;AAKCLVMKAEMNGSKLGRRAKPEGALQNNDGLYDPDCDESGLFKAKQCNGTSMCWCVNTAGVRRTDKDTEITC&#39;</span><span class="p">),</span>
                 <span class="nb">id</span><span class="o">=</span><span class="s1">&#39;P16422&#39;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;EpCAM&#39;</span><span class="p">,</span>
                 <span class="n">description</span><span class="o">=</span><span class="s1">&#39;epithelial cell adhesion molecule, 63-135&#39;</span><span class="p">)</span>
<span class="n">seq2</span> <span class="o">=</span> <span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s1">&#39;LTKCQEEVSHIPAVHPGSFRPKCDENGNYLPLQCYGSIGYCWCVFPNGTEVPNTRSRGHHNC&#39;</span><span class="p">),</span>
                 <span class="nb">id</span><span class="o">=</span><span class="s1">&#39;P04233&#39;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;HG2A&#39;</span><span class="p">,</span>
                 <span class="n">description</span><span class="o">=</span><span class="s1">&#39;HLA class II histocompatibility antigen gamma chain, 210-271&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span><span class="w"> </span><span class="nn">Bio.Align</span><span class="w"> </span><span class="kn">import</span> <span class="n">substitution_matrices</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">Bio</span><span class="w"> </span><span class="kn">import</span> <span class="n">pairwise2</span>
<span class="n">matrix</span> <span class="o">=</span> <span class="n">substitution_matrices</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="s1">&#39;BLOSUM62&#39;</span><span class="p">)</span>
<span class="n">alignments</span> <span class="o">=</span> <span class="n">pairwise2</span><span class="o">.</span><span class="n">align</span><span class="o">.</span><span class="n">globaldx</span><span class="p">(</span><span class="n">seq1</span><span class="o">.</span><span class="n">seq</span><span class="p">,</span> <span class="n">seq2</span><span class="o">.</span><span class="n">seq</span><span class="p">,</span> <span class="n">matrix</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">alignments</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[Alignment(seqA=&#39;AA--KCLVMKAE-MNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;--LTKC---Q-EEV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=99), Alignment(seqA=&#39;A-AKCLVMKAE-MNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;-LTKC---Q-EEV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=98), Alignment(seqA=&#39;-AAKCLVMKAE-MNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;LT-KC---Q-EEV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=98), Alignment(seqA=&#39;AA--KCLVMKA-EMNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;--LTKC---Q-EEV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=99), Alignment(seqA=&#39;A-AKCLVMKA-EMNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;-LTKC---Q-EEV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=98), Alignment(seqA=&#39;-AAKCLVMKA-EMNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;LT-KC---Q-EEV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=98), Alignment(seqA=&#39;AA--KCLVM-KAEMNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;--LTKC---QE-EV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=99), Alignment(seqA=&#39;A-AKCLVM-KAEMNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;-LTKC---QE-EV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=98), Alignment(seqA=&#39;-AAKCLVM-KAEMNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;LT-KC---QE-EV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=98), Alignment(seqA=&#39;AA--KCLVMKAEMNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;--LTKC--QE-EV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=98), Alignment(seqA=&#39;A-AKCLVMKAEMNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;-LTKC--QE-EV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=97), Alignment(seqA=&#39;-AAKCLVMKAEMNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C&#39;, seqB=&#39;LT-KC--QE-EV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC&#39;, score=184.0, start=0, end=97)]
</pre></div>
</div>
</div>
</div>
<p>Izpišimo še lepo eno oblikovano poravnavo:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="n">pairwise2</span><span class="o">.</span><span class="n">format_alignment</span><span class="p">(</span><span class="o">*</span><span class="n">alignments</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AA--KCLVMKAE-MNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C
    ||   . | .  |  .     |  |.  .  |  .  |  |||.|  |      ||  | |.  ||||  |  |       ||.  |       |
--LTKC---Q-EEV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC
  Score=184
</pre></div>
</div>
</div>
</div>
<p>Analogno lahko delamo tudi lokalno poravnavo, za slednjo pa pač naredimo tako:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">alignments</span> <span class="o">=</span> <span class="n">pairwise2</span><span class="o">.</span><span class="n">align</span><span class="o">.</span><span class="n">localdx</span><span class="p">(</span><span class="n">seq1</span><span class="o">.</span><span class="n">seq</span><span class="p">,</span> <span class="n">seq2</span><span class="o">.</span><span class="n">seq</span><span class="p">,</span> <span class="n">matrix</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">pairwise2</span><span class="o">.</span><span class="n">format_alignment</span><span class="p">(</span><span class="o">*</span><span class="n">alignments</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3 KCLVMKAE-MNGSK-LGRRAKPEGALQNND-GL-YD-PD-CDESG--LFKAK--QCN-GTSM--CWCV--NTAGVRRTDKDTEI--T-------C
  ||   . | .  |  .     |  |.  .  |  .  |  |||.|  |      ||  | |.  ||||  |  |       ||.  |       |
3 KC---Q-EEV--S-HI-----P--AV--H-PG-SF-RP-KCDENGNYL----PLQC-YG-SIGYCWCVFPN--G-------TEVPNTRSRGHHNC
  Score=184
</pre></div>
</div>
</div>
</div>
</section>
</section>
<hr class="docutils" />
<section id="naloga">
<h2>Naloga<a class="headerlink" href="#naloga" title="Link to this heading">#</a></h2>
<p>Poglejmo sedaj, kako se lahko z variacijo parametrov kaj naučimo. Najprej analizirajmo, ali sta poravnavi aminokislinskega zaporedja (primer zgoraj) sploh v redu. Napnite oči!</p>
<p>Če poravnavi nisva v redu predlagajte izboljšane parametre za poravnavo in sicer:</p>
<ul class="simple">
<li><p>primerno matriko,</p></li>
<li><p>kazni za vrzeli.</p></li>
</ul>
<p>Analizirajte, kako variacija parametrov (ločeno!) vpliva na poravnavo. Rezultate primerjajte s poravnavo, ki jo naredite s spletno verzijo orodja Needle na strežniku <a class="reference external" href="https://www.ebi.ac.uk/Tools/psa/">Pairwise Sequence Alignment &#64; EBI</a>.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./vaje"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="poravnava2_web.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">VAJA: Poravnava zaporedij NW &amp; SW (spletna orodja)</p>
      </div>
    </a>
    <a class="right-next"
       href="poravnava2_lastna_matrika.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">DEMO: Izračun lastne matrike in njena uporaba</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#zgledi">Zgledi</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#definicija-zaporedja">Definicija zaporedja</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#izbor-matrike-zamenjav">Izbor matrike zamenjav</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#globalna-poravnava">Globalna poravnava</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#lokalna-poravnava">Lokalna poravnava</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#globalna-in-lokalna-poravnava-aminokislinskih-zaporedij">Globalna in lokalna poravnava aminokislinskih zaporedij</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#naloga">Naloga</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Miha Pavšič
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../_static/scripts/bootstrap.js?digest=8878045cc6db502f8baf"></script>
<script defer src="../_static/scripts/pydata-sphinx-theme.js?digest=8878045cc6db502f8baf"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>